source/documentation/tools.md

```diff
@@ -0,0 +1,247 @@
+
+Instead, you should install the package locally via `install.packages()` or `remotes::install_github()`. For RShiny apps, you can add an `install.packages()` step to the `Dockerfile` to install additional packages not covered by the conda environment.yml, like so:
+
+```bash
+RUN R -e "install.packages('waffle', repos = 'https://cinc.rud.is')"
+```
+
+### JupyterLab
+
+See: <https://github.com/RobinL/cheatsheets_etc/blob/master/jupyter_conda.md>
+
+## Packrat
+
+Packrat is the most well-known package management tool for R. There's more information about it here: <https://rstudio.github.io/packrat/>
+
+It has some significant downsides. It can be quite temperamental, and difficult to debug when things go wrong - in the earlier days of the Analytical Platform, the majority of support issues related to getting Packrat working. 
+
+Furthermore, the Analytical Platform version of RStudio runs on a Linux virtual machine, and CRAN mirrors do not provide Linux compiled binaries for packages. This means that packages need to be compiled on the Analytical Platform every time they're installed, which can take a long time. This means a long wait when doing `install.packages` both in an RStudio session, and when running a Docker build for an RShiny application.
+
+### Renv
+
+[Renv](https://rstudio.github.io/renv/articles/renv.html) is a newer package billed as "Packrat 2.0". This has a number of improvements over Packrat, in the speed of download and reduction of issues of 00LOCK files that often plague Packrat. However, it is still not able to deal with OS-level dependencies, so conda is still preferred.
+
+## venv and pip
+
+When you want to use venv with Jupyter you also have to create a Jupyter kernel that links your Jupyter notebook instance to the virtual environment you created, i.e. telling it 'whenever I want to run a python command divert it through this python copy which is my virtual environment rather than the base one'.
+
+When you run `source venvname/bin/activate` in the terminal it is only making that diversion for other things that go on in the terminal (or in an IDE like VSCode it picks this cue up too), so you need to do another step in Jupyter.
+
+So say I’m starting a new project I need to do this:
+
+1. Create a virtual environment in the project folder: (terminal)
+
+    ```bash
+    python3 -m venv venvname
+    ```
+
+2. Activate the environment in the terminal to divert any terminal commands (e.g. pip installs): (terminal)
+
+    ```bash
+    source venvname/bin/activate
+    ```
+
+3. Install the module ipykernel within this venv (for creating/managing kernels for ipython which is what jupyter sits on top of): terminal (with venv activated)
+
+    ```bash
+    pip3 install ipykernel
```

Comment:
You can just use `pip` here. No need for `pip3` since "system" Python and related tools are all now Python 3 from within the `venv`.

Reply:
Good thinking. I've changed this and rewritten the tip: https://github.com/moj-analytical-services/user-guidance/pull/47/files#diff-a78a045b0c2340fd3b9edccbcf9ad2a5R212

Reply:
Just noting that I've reverted this to `pip3` again, because of this alias: https://github.com/ministryofjustice/analytics-platform-helm-charts/pull/114